lib/ansible/modules/net_tools/ldap/ldap_entry.py

```diff
@@ -145,7 +145,7 @@ def _load_attrs(self):
                 attrs[name] = []
 
             if isinstance(value, list):
-                attrs[name] = [map(to_bytes, value)]
+                attrs[name] = [to_bytes(v) for v in value]
```

Comment:
This doesn't seem to be the same thing:

```
In [1]: s = "abcdef"

In [2]: [map(str.upper, s)]
Out[2]: [['A', 'B', 'C', 'D', 'E', 'F']]

In [3]: [str.upper(v) for v in s]
Out[3]: ['A', 'B', 'C', 'D', 'E', 'F']
```

Reply:
For sure this is not the same thing. The first does not solve the issue.

Reply:
Right. I can see now what's happening. You previous fix changed this:

```diff
-attrs[name] = value
+attrs[name] = [map(to_bytes, value)]
```

Which is wrong as it creates list of list. So this PR is fixing that which of good but I would rather use this syntax than the `for` loop:

```python
attrs[name] = map(to_bytes, value)
```

Reply:
I also prefer *map* instead of a loop, but when `map(to_bytes, value)` is used, as you've asked for, the following error happens:

```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_3kcj1hb2/ansible_module_ldap_entry.py", line 241, in main
    action()
  File "/tmp/ansible_3kcj1hb2/ansible_module_ldap_entry.py", line 157, in _add
    self.connection.add_s(self.dn, modlist)
  File "/home/gcarmo/.local/lib/python3.6/site-packages/ldap/ldapobject.py", line 428, in add_s
    return self.add_ext_s(dn,modlist,None,None)
  File "/home/gcarmo/.local/lib/python3.6/site-packages/ldap/ldapobject.py", line 413, in add_ext_s
    msgid = self.add_ext(dn,modlist,serverctrls,clientctrls)
  File "/home/gcarmo/.local/lib/python3.6/site-packages/ldap/ldapobject.py", line 410, in add_ext
    return self._ldap_call(self._l.add_ext,dn,modlist,RequestControlTuples(serverctrls),RequestControlTuples(clientctrls))
  File "/home/gcarmo/.local/lib/python3.6/site-packages/ldap/ldapobject.py", line 329, in _ldap_call
    reraise(exc_type, exc_value, exc_traceback)
  File "/home/gcarmo/.local/lib/python3.6/site-packages/ldap/compat.py", line 44, in reraise
    raise exc_value
  File "/home/gcarmo/.local/lib/python3.6/site-packages/ldap/ldapobject.py", line 313, in _ldap_call
    result = func(*args,**kwargs)
ldap.PARAM_ERROR: {'desc': 'Bad parameter to an ldap routine', 'matched': 'ou=people,dc=example,dc=org'}
```

We can investigate it or just assume the solution is ok.

Reply:
Try this:

```python
attrs[name] = list(map(to_bytes, value))
```

Reply:
It worked! Just pushed.

Reply:
The same should be applied in the [`ldap_attr`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/net_tools/ldap/ldap_attr.py#L176) module.

Reply:
Could @Akasurde do it?

Reply:
You can add it into this PR.

Reply:
In which case the ldap_attr module goes wrong after the @Akasurde's PR #41248 ?

Reply:
@jtyr I've just found out how the ldap_attr module goes wrong after the @Akasurde's PR #41248. When adding a value to an attribute, the exception `TypeError: object of type 'filter' has no len()` is thrown, on line 183. The solution is `values_to_add = list(filter(self._is_value_absent, self.values))` on line 181. I have not tested the value deletion, the same error may occur. I'll make a new PR soon.

Reply:
@jtyr @Akasurde @webknjaz I've just created the PR #46818 to fix the bug mentioned above.